### PR TITLE
Remove HiveConfig binding from Delta and Iceberg with Glue 

### PR DESCRIPTION
**Pull request by findepi**
Delta Lake and Iceberg connectors do not use configuration from
`HiveConfig` and so the `HiveConfig`-provided configuration should not
be accepted when configuring the connector.

### CODE REVIEW

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/InternalDeltaLakeConnectorFactory.java [hunk 1]
```diff
@@ -135,6 +137,8 @@ public static Connector createConnector(
             Set<ConnectorTableFunction> connectorTableFunctions = injector.getInstance(Key.get(new TypeLiteral<Set<ConnectorTableFunction>>() {}));
             FunctionProvider functionProvider = injector.getInstance(FunctionProvider.class);
 
+            checkState(!injector.getBindings().containsKey(Key.get(HiveConfig.class)), "HiveConfig should not be bound");
```
Comment (losipiuk) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/InternalDeltaLakeConnectorFactory.java:140]: nit: this look over-specific - after all we are not checking other class which we do not expect to be bound if those are not.

Comment (findepi) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/InternalDeltaLakeConnectorFactory.java:140]: HiveConfig is just the most likely offender (based on history of delta connector), but i am open to suggestoins

**Comment by mosabua** [comment]
No user facing impact - so no release notes entry @findepi @losipiuk @findinpath  ?

**Comment by losipiuk** [comment]
> No user facing impact - so no release notes entry @findepi @losipiuk @findinpath ?

IIUC previously users could set config properties defined in `HiveConfig` in .properties file for iceberg/delta catalogs. No they cannot.
But those properties would have been ignored anyway so why would anyone do so.

I do not feel release notes are strictly required - for you to decide.

**Comment by mosabua** [comment]
okay ... to be on the safe side we might have to flag it as breaking change and add a release notes entry.. wdyt @martint

**Comment by mosabua** [comment]
Added RN entry.

**Comment by covalesj** [comment]
Doesn't this break the capability for iceberg connector to write to s3? IE: how do you configure writes / credentials etc for iceberg->s3 writes?

**Comment by findinpath** [comment]
>  how do you configure writes / credentials etc for iceberg-> s3 writes?

https://github.com/trinodb/trino/blob/79bc8fa09b337ac163582a708b725eb34cf36b0c/lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/HiveS3Config.java (part of `trino-hdfs` module now) 

Note that this change is about removing binding of `HiveConfig` for Delta & Iceberg connectors.

**Comment by mosabua** [comment]
@findinpath we maybe should make sure the Iceberg connector docs links to the Hive S3 docs .. after checking that all properties apply..

**Pull request merged**